### ISSUE TITLE
questions about training and validation data

### ISSUE THREAD

**Issue by puppywst** [tensorboy/pytorch_Realtime_Multi-Person_Pose_Estimation#53]
1. in train_VGG19.py, the validation also backwards the val_loss (actually also "training"), making the ReduceLROnPlateau adjust training lr on part of the  "training" data. Should validation only calculates the val_loss but not backwards loss?
2. you say "first 2644 images of val2014 marked by 'isValidation = 1', as our minival dataset."  So the openpose selected 1160 images for evaluation is all selected from those 2644 images? otherwise, I think the program has already trained on some evaluation images...

**Comment by puppywst** [comment]
in train_VGG19, function validate():
`losses.update(total_loss.item(), img.size(0))

        # compute gradient and do SGD step
        optimizer.zero_grad()
        total_loss.backward()
        optimizer.step()`

**Comment by puppywst** [comment]
@tensorboy  and I checked the image_id of those 1160 openpose images, the range is [0, 71209]. and the minival  is also in the same range. Thus 1160 images are randomly selected from 2166 images in minival.

**Comment by tensorboy** [comment]
Hi, @puppywst . I updated the `validate()` function now.

Thanks for sharing the information of validation data and minval data @puppywst :)

**Comment by puppywst** [comment]
@tensorboy  thanks for your reply.

**Comment by NokiaDDT** [comment]
Hi, @tensorboy, do you consider to replace **model.train()** to **model.eval()** in train_VGG19, function validate()?

**Comment by tensorboy** [comment]
Changed, thanks!
